ledger/ledgercore/accountresource.go

```diff
@@ -42,29 +42,34 @@ type AppResource struct {
 
 // AssignAccountResourceToAccountData assignes the Asset/App params/holdings contained
 // in the AccountResource to the given basics.AccountData, creating maps if necessary.
-func AssignAccountResourceToAccountData(cindex basics.CreatableIndex, resource AccountResource, ad *basics.AccountData) {
+func AssignAccountResourceToAccountData(cindex basics.CreatableIndex, resource AccountResource, ad *basics.AccountData) (assigned bool) {
```

Comment:
maybe a comment like 
```suggestion
// Returns true if the AccountResource contained a new or updated resource, and false if the AccountResource contained no changes (indicating the resource was deleted).
func AssignAccountResourceToAccountData(cindex basics.CreatableIndex, resource AccountResource, ad *basics.AccountData) (assigned bool) {
```

Reply:
I guess another way to do it would be to have a method like
```
// IsEmpty is not a great name, something clearer
func (r *AccountResource) IsEmpty() bool {
   return r.AssetParams == nil && r.AssetHolding == nil && r.AppLocalState == nil && r.AppParams == nil
}

Reply:
I thought about checking in `addResource` but I think `AssignAccountResourceToAccountData` is a right place because 1) it has "insertion" logic 2) it basically creates the output data structure 3) it is used only in `addResource`.
Updated the `AssignAccountResourceToAccountData` comment as suggested.